COM-73 Set up user roles for pilot

prashantsable1991 opened the pull request:
Added user roles for pilot.

prashantsable1991 committed:
COM-73 Set up user roles for pilot

tmvumbi2 commented on sql/create_users_and_user_roles.sql:
```diff
@@ -0,0 +1,92 @@
+INSERT IGNORE INTO person (uuid,gender,date_created,creator) VALUES ('3f64faba-ad31-11e9-a2a3-2a2ae2dbcce4','F',NOW(),('4'));
+INSERT IGNORE INTO person_name (person_id,uuid,given_name,family_name) 
+VALUES
+((SELECT person_id FROM person WHERE uuid = '3f64faba-ad31-11e9-a2a3-2a2ae2dbcce4'),(uuid()),('Nurse'),(''));
```
@prashantsable1991 please replace `uuid()` with a hard coded value (please see line 11 and 15 as well)

tmvumbi2 replied:
@prashantsable1991 why inserted values are between parentheses?

tmvumbi2 commented on sql/create_users_and_user_roles.sql:
```diff
@@ -0,0 +1,92 @@
+SET @nursesalt = (SHA2(SUBSTRING(MD5(RAND()), -10), 512));
+SET @doctorsalt = (SHA2(SUBSTRING(MD5(RAND()), -10), 512));
+SET @receptionsalt = (SHA2(SUBSTRING(MD5(RAND()), -10), 512));
```
@prashantsable1991 please add line breaks between queries for a better readability

tmvumbi2 commented on sql/create_users_and_user_roles.sql:
```diff
@@ -0,0 +1,92 @@
+((SELECT person_id FROM person WHERE uuid = '3f64fe70-ad31-11e9-a2a3-2a2ae2dbcce4'),'4',NOW(),'1ad6adf0-adfa-11e9-a2a3-2a2ae2dbcce4');
+INSERT IGNORE INTO user_role (user_id,role)
+VALUES 
+((SELECT user_id FROM users WHERE username = 'Nurse'),
```
@prashantsable1991 please create a variable that stores the result of `SELECT user_id FROM users WHERE username = 'Nurse'` (this will prevent repeating the queries). This can be done for Doctor and Reception as well

tmvumbi2 committed:
COM-73 Add line breaks between queries

Adds line breaks between queries to make the file more readable

tmvumbi2 committed:
COM-73 Store intermediary results into variables

This avoid redundant queries

tmvumbi2 committed:
COM-73 Hard code new uuids

So that they are the same across all the environments, and easy to reference in future

tmvumbi2 committed:
COM-73 Remove extra parentheses

Removes extra parentheses

tmvumbi2 committed:
COM-73 Add indentations to the queries

Adds indentations to the queries to make them more readable

tmvumbi2 committed:
COM-73 Remove queries retrieving the uuids

The person uuids are already hard coded, thus there is no need to query and retrieve them. 
This commit also updates queries that retrieve the user ids to filter by uuid instead username.

tmvumbi2 committed:
COM-73 Add spaces after commas

To make the queries more readable

tmvumbi2 commented:
@prashantsable1991 the column `uuid` is not unique in the table `users`;  
the side effect, unfortunately, is that `INSERT IGNORE ...` will not work in the case of the users table; resulting in several users created with the same uuid every time the file is executed. 

We'll need to think of a way to prevent inserting a new user if the uuid we want to add already exists, see: https://stackoverflow.com/questions/913841/mysql-conditional-insert

```mysql> DESCRIBE users;
+-----------------+--------------+------+-----+---------+----------------+
| Field           | Type         | Null | Key | Default | Extra          |
+-----------------+--------------+------+-----+---------+----------------+
| user_id         | int(11)      | NO   | PRI | NULL    | auto_increment |
| system_id       | varchar(50)  | NO   |     |         |                |
| username        | varchar(50)  | YES  |     | NULL    |                |
| password        | varchar(128) | YES  |     | NULL    |                |
| salt            | varchar(128) | YES  |     | NULL    |                |
| secret_question | varchar(255) | YES  |     | NULL    |                |
| secret_answer   | varchar(255) | YES  |     | NULL    |                |
| creator         | int(11)      | NO   | MUL | 0       |                |
| date_created    | datetime     | NO   |     | NULL    |                |
| changed_by      | int(11)      | YES  | MUL | NULL    |                |
| date_changed    | datetime     | YES  |     | NULL    |                |
| person_id       | int(11)      | NO   | MUL | NULL    |                |
| retired         | tinyint(1)   | NO   |     | 0       |                |
| retired_by      | int(11)      | YES  | MUL | NULL    |                |
| date_retired    | datetime     | YES  |     | NULL    |                |
| retire_reason   | varchar(255) | YES  |     | NULL    |                |
| uuid            | char(38)     | NO   |     | NULL    |                |
+-----------------+--------------+------+-----+---------+----------------+```

tmvumbi2 committed:
COM-73 Insert users only if they do not exist already

This commit adds logic to verify that a user does not exist before adding a new record. This is necessary because the column users.uuid does not have a unique constraint (therefore we cannot rely on the built in constraints to avoid duplicates)
